src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,64 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.CommandResult;
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+import java.nio.file.Path;
+
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Initiate the Duke bot.
+     * create UI, duke.storage.Storage, duke.task.TaskList.
+     * @param filePath filePath stores the txt file used to load data.
+     */
+    public Duke(Path filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+
+    /**
+     * Chat bot function, run infinitely until user types "bye".
+     */
+    public void runChatBot() {
+        this.ui.welcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                Command c = Parser.parseMessage(fullCommand);
+                CommandResult commandResult = c.execute(tasks, ui, storage);
+                isExit = c.getIsExit();
+            } catch (DukeException e) {
+                ui.showError(e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Prints a logo of the name Duke.
+     * @param args Set up main function
+     */
+    public static void main(String[] args) {
+        String home = System.getProperty("user.home");
+        Path path = java.nio.file.Paths.get(home, "Desktop", "GitHub", "duke","data","data.txt");
```

Comment:
might wanna keep this as generic as possible! Would suggest creating ./data in the pwd.